packages/node/src/node/NodePhysicalProperties.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * @license
+ * Copyright 2022-2025 Matter.js Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { NetworkCommissioningClient } from "#behaviors/network-commissioning";
+import { PowerSourceClient } from "#behaviors/power-source";
+import { ThreadNetworkDiagnosticsClient } from "#behaviors/thread-network-diagnostics";
+import { PowerSource } from "#clusters/power-source";
+import { ThreadNetworkDiagnostics } from "#clusters/thread-network-diagnostics";
+import { Endpoint } from "#endpoint/index.js";
+import { DeviceClassification, IcdManagement } from "#model";
+import { Node } from "#node/Node.js";
+import { PhysicalDeviceProperties } from "#protocol";
+import { ClusterId } from "#types";
+
+/**
+ * Inspects a node to generate {@link PhysicalDeviceProperties}.
+ */
+export function NodePhysicalProperties(node: Node) {
+    const rootEndpointServerList = [...node.state.descriptor.serverList];
+
+    const properties: PhysicalDeviceProperties = {
+        threadConnected: false,
+        wifiConnected: false,
+        ethernetConnected: false,
+        rootEndpointServerList,
+        isBatteryPowered: false,
+        isIntermittentlyConnected: rootEndpointServerList.includes(IcdManagement.id as ClusterId),
+        isThreadSleepyEndDevice: false,
+    };
+
+    inspectEndpoint(node, properties);
+
+    return properties;
+}
+
+function inspectEndpoint(endpoint: Endpoint, properties: PhysicalDeviceProperties) {
+    // Network interface support
+    const network = endpoint.behaviors.typeFor(NetworkCommissioningClient);
+    if (network) {
+        const features = network.schema.supportedFeatures;
+        if (features.has("WI")) {
+            properties.wifiConnected = true;
+        }
+        if (features.has("TH")) {
+            properties.threadConnected = true;
+        }
+        if (features.has("ET")) {
```

Comment:
We should consider it "ethernetconnected" when there is no Network cluster existing, then it is "connected to network already, not managed by matter" ...

Reply:
I figured the ethernet flag should be for ethernet and the "not managed by matter" bit comes from having no flags set

Reply:
or we rename from "ethernet" to "somehow on network" because in the end such devices are usually wired or such

Reply:
I vote we leave `isEthernetConnected` for known ethernet and leave `isEthernetConnected === isWifiConnected === isThreadConnected === false` to indicate "somehow on network".  Could add an `isConnectedMystically` flag but I'm not sure there's a benefit